File: mapping/prep_data.py
import pandas as pd
import duckdb as db

def get_df_from_db():
    """
    Read in the TSV file containing the feature data and return the 
    results of the query as a Pandas DataFrame
    """

    # Ingest local authority data
    lad_data = db.read_csv(
    "data/median_annual_pay.tsv", 
    header=True,
    sep="\t")

    # Query local authority data and output to DataFrame
    df = db.sql(
    """
    SELECT date AS year, 
    "local authority: district / unitary (as of April 2023)" AS lad, 
    TRY_CAST(value AS INTEGER) AS median_ann_pay
    FROM lad_data
    WHERE "value type"='number'
    """
    ).df()

    return df

File: mapping/prep_data.R
library(magrittr)

get_df_from_db <- function() {
    df <- read.delim(file = "data/median_annual_pay.tsv", na.strings = "")
    
    df <- dplyr::select(df, date, local.authority..district...unitary..as.of.April.2023., value) |>
        dplyr::rename(year = date, lad = local.authority..district...unitary..as.of.April.2023., median_ann_pay = value) |>
        dplyr::filter(!grepl("[^0-9]", median_ann_pay), !is.na(median_ann_pay)) |>
        data.frame()
    
    df$median_ann_pay <- as.numeric(df$median_ann_pay)
    df$year <- as.numeric(year)
    return(df)
    
}
    